security-jpa-quickstart failing in native

The native build of the security-jpa-quickstart is failing with the following errors.

Example of a build failing: https://github.com/quarkusio/quarkus-quickstarts/runs/3356862727?check_suite_focus=true .

We need this fixed before the 2.2.0.Final release.

```
Error:  org.acme.elytron.security.jpa.JpaSecurityRealmTestIT.shouldAccessAdminWhenAdminAuthenticated  Time elapsed: 0.046 s  <<< FAILURE!
java.lang.AssertionError: 
1 expectation failed.
Expected status code <200> but was <401>.


Error:  org.acme.elytron.security.jpa.JpaSecurityRealmTestIT.shouldNotAccessUserWhenAdminAuthenticated  Time elapsed: 0.014 s  <<< FAILURE!
java.lang.AssertionError: 
1 expectation failed.
Expected status code <403> but was <401>.


Error:  org.acme.elytron.security.jpa.JpaSecurityRealmTestIT.shouldAccessUserAndGetIdentityWhenUserAuthenticated  Time elapsed: 0.012 s  <<< FAILURE!
java.lang.AssertionError: 
1 expectation failed.
Expected status code <200> but was <401>.
```

--- a/extensions/elytron-security-common/deployment/src/main/java/io/quarkus/elytron/security/common/deployment/QuarkusSecurityCommonProcessor.java
+++ b/extensions/elytron-security-common/deployment/src/main/java/io/quarkus/elytron/security/common/deployment/QuarkusSecurityCommonProcessor.java
@@ -23,6 +23,17 @@ public class QuarkusSecurityCommonProcessor {
     }
 
     /**
+     * Graal VM now seems to loose providers registered at static init
+     *
+     * We re-register at runtime (which is a no-op in JVM mode)
+     */
+    @BuildStep
+    @Record(ExecutionTime.RUNTIME_INIT)
+    public void registerPasswordProviderForNative(ElytronCommonRecorder recorder) {
+        recorder.registerPasswordProvider();
+    }
+
+    /**
      * Register the Elytron-provided password factory SPI implementation
      *
      * @param classes producer factory for ReflectiveClassBuildItems

